account_statement_import_sheet_file/models/account_statement_import_sheet_parser.py

```diff
@@ -0,0 +1,497 @@
+# Copyright 2019 ForgeFlow, S.L.
+# Copyright 2020 CorporateHub (https://corporatehub.eu)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import itertools
+import logging
+import math
+import re
+from collections.abc import Iterable
+from datetime import datetime
+from decimal import Decimal
+from io import StringIO
+from os import path
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from csv import reader
+
+    import xlrd
+    from xlrd.xldate import xldate_as_datetime
+except (OSError, ImportError) as err:  # pragma: no cover
+    _logger.error(err)
+
+try:
+    import chardet
+except ImportError:
+    _logger.warning(
+        "chardet library not found, please install it "
+        "from http://pypi.python.org/pypi/chardet"
+    )
+
+
+class AccountStatementImportSheetParser(models.TransientModel):
+    _name = "account.statement.import.sheet.parser"
+    _description = "Bank Statement Import Sheet Parser"
+
+    @api.model
+    def parse_header(self, csv_or_xlsx, mapping):
+        if mapping.no_header:
+            return []
+        header_line = mapping.header_lines_skip_count
+        # prevent negative indexes
+        if header_line > 0:
+            header_line -= 1
+        if isinstance(csv_or_xlsx, tuple):
+            header = [
+                str(value).strip() for value in csv_or_xlsx[1].row_values(header_line)
+            ]
+        else:
+            [next(csv_or_xlsx) for _i in range(header_line)]
+            header = [value.strip() for value in next(csv_or_xlsx)]
+        if mapping.offset_column:
+            header = header[mapping.offset_column :]
+        return header
+
+    @api.model
+    def parse(self, data_file, mapping, filename):
+        journal = self.env["account.journal"].browse(self.env.context.get("journal_id"))
+        currency_code = (journal.currency_id or journal.company_id.currency_id).name
+        account_number = journal.bank_account_id.acc_number
+
+        lines = self._parse_lines(mapping, data_file, currency_code)
+        if not lines:
+            return currency_code, account_number, [{"transactions": []}]
+
+        lines = list(sorted(lines, key=lambda line: line["timestamp"]))
+        first_line = lines[0]
+        last_line = lines[-1]
+        data = {
+            "date": first_line["timestamp"].date(),
+            "name": _("%(code)s: %(filename)s")
+            % {
+                "code": journal.code,
+                "filename": path.basename(filename),
+            },
```

Comment:
Instead of passing the arguments as format string, pass them as parameters.
```suggestion
            "name": self.env._("%(code)s: %(filename)s", code=journal.code, filename=path.basename(filename))
```